discord/channel.py

```diff
@@ -3189,7 +3189,7 @@ def __init__(
     ):
         self._state: ConnectionState = state
         self._channel: Object = Object(id=id)
-        self.id: int = id
+        self.id: int = int(id)
```

Comment:
Why not change the part where the ID is passed in with the wrong type?

Reply:
I suppose, but shouldn't we guarantee all IDs are int for parity with full channel objects?

Reply:
The library usually converts before passing in, so I think we should continue that